robottelo/api/utils.py

```diff
@@ -213,3 +217,78 @@ def get_role_by_bz(bz_id):
                 ]
             entities.Filter(permission=permissions, role=role).create()
     return role.read()
+
+
+def create_role_permissions(role, permissions_types_names):  # pragma: no cover
+    """Create role permissions found in dict permissions_types_names.
+
+    :param role: nailgun.entities.Role
+    :param permissions_types_names: a dict containing resource types
+        and permission names to add to the role, example usage.
+
+          ::
+
+           permissions_types_names = {
+               None: ['access_dashboard'],
+               'Organization': ['view_organizations'],
+               'Location': ['view_locations'],
+               'Katello::KTEnvironment': [
+                   'view_lifecycle_environments',
+                   'edit_lifecycle_environments',
+                   'promote_or_remove_content_views_to_environments'
+               ]
+           }
+           role = entities.Role(name='example_role_name').create()
+           create_role_permissions(role, permissions_types_names)
+    """
+    for resource_type, permissions_name in permissions_types_names.items():
+        if resource_type is None:
+            permissions_entities = []
+            for name in permissions_name:
+                result = entities.Permission(name=name).search()
+                if not result:
+                    raise APIError('permission "{}" not found'.format(name))
```

Comment:
Imo `nailgun.entities.APIResponseError` would match better here, no need to define new exception. Same for all the rest exceptions.